apps/game_backend/lib/game_backend/rewards.ex

```diff
@@ -43,4 +44,73 @@ defmodule GameBackend.Rewards do
         insert_afk_reward_rate(%{user_id: user_id, currency_id: currency_id, rate: rate_increment})
     end
   end
+
+  @doc """
+  Receives a user.
+  Returns the next daily reward for the given user.
+  Or the first claim if never claimed at all.
+  """
+  def claim_daily_reward(%GameBackend.Users.User{last_daily_reward_claim_at: nil}),
+    do: claim_first_reward(Utils.get_daily_rewards_config())
+
+  def claim_daily_reward(user) do
+    daily_reward_config = Utils.get_daily_rewards_config()
+    yesterday = DateTime.utc_now() |> Date.add(-1)
+
+    case Date.compare(user.last_daily_reward_claim_at, yesterday) do
+      :eq -> claim_next_reward(daily_reward_config, user.last_daily_reward_claim)
+      _ -> claim_first_reward(daily_reward_config)
+    end
+  end
+
+  defp claim_first_reward(daily_reward_config) do
+    case Map.get(daily_reward_config, "1") do
+      nil -> {:error, :invalid_reward}
+      first_daily_reward -> {:ok, first_daily_reward}
+    end
+  end
+
+  defp claim_next_reward(daily_reward_config, current_reward) do
+    case Map.get(daily_reward_config, current_reward) do
+      nil -> {:error, :invalid_reward}
+      next_daily_reward -> {:ok, next_daily_reward}
+    end
+  end
+
+  @doc """
+  Receives a user.
+  Returns {:ok, can_claim} if the user claimed today already or never claimed at all.
+  """
+  def user_claimed_today(%GameBackend.Users.User{last_daily_reward_claim_at: nil}), do: {:ok, :can_claim}
+
+  def user_claimed_today(user) do
+    now = DateTime.utc_now()
+
+    case Date.compare(user.last_daily_reward_claim_at, now) do
+      :eq -> {:ok, :can_claim}
```

Comment:
Is this correct? Shouldn't this be `:lt` so it says how you haven't claimed today

Reply:
Yeah you're right, fixed now.